public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  repeating that `<div>` with `*ngFor`.
+  
+  The trick lies in knowing how to write the `*ngFor`. There are three key points:
+
+  1. Add another wrapping `<div>`, around the `<div>` with `*ngFor`, and 
+  set its `formArrayName` directive to `"secretLairs"`.
+  This step establishes the _secretLairs_ `FormArray` as the context for form controls in the inner, repeated HTML template. 
+
+  1. The source of the repeated items is the `FormArray.controls`, not the `FormArray` itself. 
+  Each control is an _address_ `FormGroup`, exactly what the previous (now repeated) template HTML expected.
+
+  1. Each repeated `FormGroup` needs a unique `FormGroupName` which must be the index of the `FormGroup` in the `FormArray`.
+  You'll re-use that index to compose a unique label for each address.
+
+  Here's the skeleton for the _secret lairs_ section of the HTML template:
++makeExample('reactive-forms/ts/app/hero-detail-5.component.html', 'form-array-skeleton','app/hero-detail.component.html (*ngFor)')(format=".")
+
+:marked
+  Here's the complete template for the _secret lairs_ section:
++makeExample('reactive-forms/ts/app/hero-detail-5.component.html', 'form-array','app/hero-detail.component.html (excerpt)')
+
+:marked
+  ### Add a new lair to the _FormArray_
+
+  Add an `addLair` method that gets the _secretLairs_ `FormArray` and appends a new _address_ `FormGroup` to it. 
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'add-lair','app/hero-detail.component.ts (addLair method)')(format=".")
+
+:marked
+  Place a button on the form so the user can add a new _secret lair_ and wire it to the component's `addLair` method.
+
++makeExample('reactive-forms/ts/app/hero-detail-5.component.html', 'add-lair','app/hero-detail.component.html (addLair button)')(format=".")
+
+.alert.is-important
+  :marked
+    Be sure to **add the `type="button"` attribute**. 
+    In fact, you should always specify a button's `type`.
+    Without an explict type, the button type defaults to "submit".
+    When you later add a _form submit_ action, every "submit" button triggers the submit action which
+    might do something like save the current changes.
+    You do not want to save changes when the user clicks the _Add a Secret Lair_ button.
+
+:marked
+  ### Try it!
+
+  Back in the browser, select the hero named "Magneta".
+  "Magneta" doesn't have an address, as you can see in the diagnostic JSON at the bottom of the form.
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/addresses-array.png" width="400px" alt="JSON output of addresses array")
+
+:marked
+  Click the "_Add a Secret Lair_" button. 
+  A new address section appears. Well done!
+
+  ### Remove a lair
+
+  This example can _add_ addresses but it can't _remove_ them.
+  For extra credit, write a `removeLair` method and wire it to a button on the repeating address HTML.
+
+.l-main-section
+a#watch-control
+:marked
+  ## Watch control changes
```

Comment:
I can see your point there.  I'm okay with using watch in the paragraph. But what do you think about adding a quick point that "watch" does not mean what it did in Angular 1?